src/trunk.c

```diff
@@ -8964,39 +8965,52 @@ trunk_config_init(trunk_config        *trunk_cfg,
    // Inline what we would get from trunk_pivot_size(trunk_handle *).
    trunk_pivot_size = data_cfg->max_key_size + sizeof(trunk_pivot_data);
 
-   // Setting hard limit and over overprovisioning
+   // Setting hard limit and check configuration for over-provisioning
    trunk_cfg->max_pivot_keys = trunk_cfg->fanout + TRUNK_EXTRA_PIVOT_KEYS;
    uint64 header_bytes       = sizeof(trunk_hdr);
 
    uint64 pivot_bytes = (trunk_cfg->max_pivot_keys
                          * (data_cfg->max_key_size + sizeof(trunk_pivot_data)));
    uint64 branch_bytes =
       trunk_cfg->max_branches_per_node * sizeof(trunk_branch);
-   uint64 trunk_node_min_size = header_bytes + pivot_bytes + branch_bytes;
-   uint64 available_pivot_bytes =
-      cache_config_page_size(cache_cfg) - header_bytes - branch_bytes;
+   uint64 trunk_node_min_size   = header_bytes + pivot_bytes + branch_bytes;
+   uint64 page_size             = trunk_page_size(trunk_cfg);
+   uint64 available_pivot_bytes = page_size - header_bytes - branch_bytes;
    uint64 available_bytes_per_pivot =
       available_pivot_bytes / trunk_cfg->max_pivot_keys;
-   uint64 available_bytes_per_pivot_key =
-      available_bytes_per_pivot - sizeof(trunk_pivot_data);
-   platform_assert(trunk_node_min_size < cache_config_page_size(cache_cfg),
-                   "\nTrunk node does not fit in page size as configured.\n"
-                   "node->hdr: %luB\n"
-                   "pivots: %luB (max_pivot=%lu x %luB)\n"
-                   "branches %luB (max_branches=%lu x %luB).\n"
-                   "Maximum key size supported with current config: %luB.\n",
-                   header_bytes,
-                   pivot_bytes,
-                   trunk_cfg->max_pivot_keys,
-                   data_cfg->max_key_size + sizeof(trunk_pivot_data),
-                   branch_bytes,
-                   max_branches_per_node,
-                   sizeof(trunk_branch),
-                   available_bytes_per_pivot_key);
+
+   // Deal with mis-configurations where we don't have available bytes per
+   // pivot key
+   uint64 available_bytes_per_pivot_key = 0;
+   if (available_bytes_per_pivot > sizeof(trunk_pivot_data)) {
```

Comment:
The error in mis-reported 'max key size' value creeps in here, due to an overflow of the `uint64` variable.